### ISSUE TITLE
import self issue bug 

### ISSUE THREAD

**Issue by bluelovers** [TypeStrong/ts-node#648]
## file

./strtable.ts
```ts
import * as self from './strtable';
export default self;
console.log(__filename, require.extensions);
```

## cli

```
ts-node strtable.ts
```

#### Expected behavior

should only once , even if .js exists

```
strtable.ts {
  '.ts': [Function],
  '.tsx': [Function],
  '.js': [Function],
  '.json': [Function],
  '.node': [Function] }
```

#### Actual behavior:

will load self x2 ( .js + .ts )

```
strtable.js { '.js': [Function],
  '.json': [Function],
  '.node': [Function],
  '.ts': [Function],
  '.tsx': [Function] }

strtable.ts { '.js': [Function],
  '.json': [Function],
  '.node': [Function],
  '.ts': [Function],
  '.tsx': [Function] }
```

**Comment by blakeembrey** [comment]
Can you create a reproduction I can look at? There's no way, from this information you've given me, such behaviour could be replicated.

**Comment by bluelovers** [comment]
![2018-07-29-10-44-59-6](https://user-images.githubusercontent.com/167966/43362432-9f919cb6-931c-11e8-8dd6-d4cd96b430c2.png)

![2018-07-29-10-45-18-7](https://user-images.githubusercontent.com/167966/43362433-a1d64f1c-931c-11e8-8304-a6be609bf541.png)

**Comment by bluelovers** [comment]
when we use ts-node and import , should use .ts > .js

**Comment by blakeembrey** [comment]
I disagree. Messing with core node.js expected behaviour is not something I intend to support, and having a behaviour that's different in `ts-node` vs `tsc` + node.js goes against the mission of this module.

**Comment by blakeembrey** [comment]
That’s still altering the behaviour of node somewhat, I would need to override the module resolution logic. Plus it wouldn’t solve anything really for you - node always loads Javascript first, so you’re just pushing the issue to the second file that’s loaded. You can replicate this by putting a file before your “self import” and it’ll show it only loads the JavaScript and never TypeScript.

**Comment by bluelovers** [comment]
how about make option?

https://github.com/TypeStrong/ts-node/pull/649

**Comment by blakeembrey** [comment]
But why?

**Comment by bluelovers** [comment]
make sure when we run .ts is new

sometime when .js, not update done or watcher fail for watch and make new .js

**Comment by blakeembrey** [comment]
I still don’t understand why. Can’t you just output JS to a different directory? 

I think this has too many negative side effects. For example, this will create an issue for anyone writing JS libraries in TypeScript and publishing them like you are (JS next to TS). The TypeScript will be resolved first and, as a result, won’t be able to compile any anyone using this flag might run into this random error, creating more issues for myself to triage.

**Comment by bjg2** [comment]
I want to point out that having `.js` > `.json` > `.ts` > `.tsx` order also does mess node expected execution, as mentioned in https://github.com/TypeStrong/ts-node/issues/786#issuecomment-465721702. I think correct ordering for ts-node should be: `.ts` > `.tsx` > `.js` > `.json`.
